python/cuxfilter/charts/core/aggregate/core_aggregate.py

```diff
@@ -16,10 +107,273 @@ def _compute_array_all_bins(
         update_data_x: updated_data_x, np.array()
         update_data_y: updated_data_x, np.array()
         """
-        result_array = np.zeros(shape=(int(source_x.max()),))
-        # -1 for 0-based indexing
-        np.put(result_array, update_data_x.astype(int) - 1, update_data_y)
-        return result_array[source_x.astype(int) - 1]
+        result_array = np.zeros(shape=source_x.shape)
+        indices = [np.where(x_ == source_x)[0][0] for x_ in update_data_x]
+        np.put(result_array, indices, update_data_y)
+        return result_array
+
+    def compute_min_max(self, dashboard_cls):
+        if type(dashboard_cls._cuxfilter_df.data) == dask_cudf.core.DataFrame:
+            self.min_value = (
+                dashboard_cls._cuxfilter_df.data[self.x].min().compute()
+            )
+            self.max_value = (
+                dashboard_cls._cuxfilter_df.data[self.x].max().compute()
+            )
+        else:
+            self.min_value = dashboard_cls._cuxfilter_df.data[self.x].min()
+            self.max_value = dashboard_cls._cuxfilter_df.data[self.x].max()
+
+    def compute_stride(self):
+        self.stride_type = dt.transform_stride_type(
+            self.stride_type, self.x_dtype
+        )
+
+        if self.stride_type == int and self.max_value < 1:
+            self.stride_type = float
+
+        if self.stride is None and self.data_points is not None:
+            if self.stride_type == int:
+                self.stride = self.stride_type(
+                    round((self.max_value - self.min_value) / self.data_points)
+                )
+            else:
+                self.stride = self.stride_type(
+                    (self.max_value - self.min_value) / self.data_points
+                )
```

Comment:
or even:
```python
            raw_stride = (self.max_value - self.min_value) / self.data_points
            stride = round(raw_stride) if self.stride_type == int else raw_stride
            self.stride = self.stride_type(stride)
```
Just to be clear I am *not* advocating purely for shorter / more compressed code just for the sake of being shorter, but for isolating individual steps and doing them once, instead of spreading over multiple codepaths. In this case that is:

1. raw stride value
1. round (or not)
1. set from stride type

Reply:
makes sense, thanks for the suggestion. Implemented it.